### DISCUSSION TITLE
Cache does not update

### DISCUSSION THREAD

**Discussion by joaolandino** [vercel/next.js#51738] [Help]
### Summary

Hey guys!

I have a connection with postgres using prism and, after updating data in a form, when I go to another screen and return to the form, the old data persists.

Example of my page.tsx

```
import prismadb from '@/app/lib/prismadb'

export const revalidate = 0

async function getPosts() {
  const page = await prismadb.page.findFirst({
    select: {
      id: true,
      path: true,
      imageUrl: true,
    },
  })

  return {
    page: page,
  }
}

export default async function InfosPage() {
  const posts = await getPosts()

  return (
    <>
      <p></p>
    </>
  )
}

```

Every time I land on this screen, the data is still cached, and when I look in the database the data is up to date.

Would anyone know how to help me please?

**Comment by masterbater** [comment]
https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config#dynamic

**Reply by joaolandino** [reply]
Still not working :(
Whenever I access the page, update the information in the <FormContainer /> form, leave the page, and return to the page again, the data continues as before.

My code is:
```
import { FormContainer } from '@/app/components/FormContainer'
import { authOptions } from '@/app/lib/auth'
import prismadb from '@/app/lib/prismadb'
import { getServerSession } from 'next-auth'

export const dynamic = 'force-dynamic'

async function getPosts() {
  const session = await getServerSession(authOptions)

  const page = await prismadb.page.findFirst({
    select: {
      id: true,
      path: true,
      imageUrl: true,
    },
    where: {
      userId: {
        equals: session?.user?.id,
      },
    },
  })

  return {
    page: page
  }
}

export default async function InfosPage() {
  const pageInfos = await getPosts()

  return (
    <>
      <FormContainer page={pageInfos} />
    </>
  )
}

```

I record a demo

https://app.usebubbles.com/q6q5RGVzVYKs1Mb6XV63po

**Reply by masterbater** [reply]
Try this https://nextjs.org/docs/app/building-your-application/data-fetching/revalidating#background-revalidation
or read this
https://nextjs.org/docs/app/building-your-application/rendering/static-and-dynamic-rendering

**Reply by mirik999** [reply]
export const dynamic = 'force-dynamic'  this code should force fetching the page on every request and get latest data. But it behavies like cached. 
also const dynamic = 'force-dynamic'  and revalidate = 0 does the same thing. None of them helps to avoid page caching.

**Comment by mirik999** [comment]
The same problem , also looking for a solution. 
export const dynamic = 'force-dynamic' - does not disable the cache system. 
Network tab in browser also does not show fetching the page after first fetch.

**Comment by mirik999** [comment]
@joaolandino Heyo , if you have not found a solution so look at the solution from **EugeneMeles** https://github.com/vercel/next.js/issues/42991.
